modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -523,29 +555,30 @@ GDScriptParser::DataType GDScriptAnalyzer::resolve_datatype(GDScriptParser::Type
 				break;
 			}
 			if (script_class->members_indices.has(first)) {
-				GDScriptParser::ClassNode::Member member = script_class->members[script_class->members_indices[first]];
+				GDScriptParser::ClassNode::Member &member = script_class->members.write[script_class->members_indices[first]];
+				resolve_class_member(script_class, member);
 				switch (member.type) {
 					case GDScriptParser::ClassNode::Member::CLASS:
-						result = member.m_class->get_datatype();
+						result = member.get_datatype();
 						found = true;
 						break;
 					case GDScriptParser::ClassNode::Member::ENUM:
-						result = member.m_enum->get_datatype();
+						result = member.get_datatype();
 						found = true;
 						break;
```

Comment:
You can now let `CLASS` fallthrough to `ENUM` since the code is the same! Wonder if there's any other places where this could be done :)